### ISSUE TITLE
Cross-provider connection

### ISSUE THREAD

**Issue by miha-plesko** [ManageIQ/manageiq-providers-nuage#131]
Suppose we have **OpenStack** cloud provider plus **Nuage** network provider. OpenStack cloud provider inventories virtual machines while Nuage network provider inventories NetworkPorts.

Here comes the catch: Nuage NetworkPort knows VM ems_ref of the OpenStack VM. We want to be able to connect the two. In other words: we want Nuage NetworkPort to be connected to OpenStack VM.

During inventoring, we can assume OpenStack VM is already in VMDB. When Nuage inventoring occurs, we get ems_ref of OpenStack's VM upon NetworkPort inventoring and we want to set foreign key like this:

```
network_port.device = Vm.find_by(:ems_ref => data['openstack_ems_ref'])
--- Nuage ---------   ------------------ OpenStack VM -----------------
```

@Ladas @agrare in theory this should work, right?

RFE BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1619544

**Comment by Ladas** [comment]
@miha-plesko yes, just we should define :vms with strategy :local_db_find_references, then you will take it as any inventory collection and just do lazy find

so something like

```
include ManageIQ::Providers::Amazon::Inventory::Persister::Definitions::CloudCollections

def initialize_inventory_collections
    ... your persister content

      add_collection(cloud, :vms) do |builder|
        builder.add_properties(
          :arel        => Vm.all
          :strategy => :local_db_find_references
        )
      end
end   
```

That will work if :ems_ref is unique for all managers added

**Comment by miha-plesko** [comment]
Thanks so much, you sure spared me days with this snippet.
